### PR TITLE
Sketch distribute fix (#1017)

### PR DESCRIPTION
**Pull request by lorenzncode**
* distribute was creating an extra location
* Split distribute, rarray, parray tests

### CODE REVIEW

**Comment by lorenzncode** [comment]
Marking as draft for now as this change currently also implements a nested function which I think may be frowned upon.

**Comment by adam-urbanczyk** [comment]
@lorenzncode could you merge master into this?

File: cadquery/sketch.py [hunk 1]
```diff
@@ -386,18 +387,42 @@ def distribute(
...
         if not self._selection:
             raise ValueError("Nothing selected to distribute over")
 
-        params = [start + i * (stop - start) / n for i in range(n + 1)]
+        if 1 - abs(stop - start) < TOL:
+            trimmed = False
+        else:
+            trimmed = True
+
+        params = (
```
Comment (adam-urbanczyk) [inline cadquery/sketch.py:401]: Could you split `params` into two variables?

**Review by adam-urbanczyk — CHANGES_REQUESTED** [review]
Thanks, LGTM with a minor remark

File: cadquery/sketch.py (hunk 1 above)
Comment (lorenzncode) [reply cadquery/sketch.py]: Done.  Thanks for reviewing.

**Review by adam-urbanczyk — APPROVED** [review]

**Review by jmwright — APPROVED** [review]
Looks good. Thanks for all your work on this @lorenzncode !

**Pull request merged**